bin/propolis-server/src/lib/stats.rs

```diff
@@ -4,113 +4,125 @@
 
 //! Methods for starting an Oximeter endpoint and gathering server-level stats.
 
+use std::net::SocketAddr;
+use std::sync::{Arc, Mutex};
+
 use omicron_common::api::internal::nexus::ProducerEndpoint;
 use omicron_common::api::internal::nexus::ProducerKind;
 use oximeter::{
-    types::{Cumulative, ProducerRegistry, Sample},
-    Metric, MetricsError, Producer,
+    types::{ProducerRegistry, Sample},
+    MetricsError, Producer,
 };
-use oximeter_producer::{Config, Error, Server};
+use oximeter_instruments::kstat::KstatSampler;
+use oximeter_producer::{Config, Server};
+use propolis_api_types::InstanceProperties;
 use slog::Logger;
 
-use std::net::SocketAddr;
-use std::sync::{Arc, Mutex};
-use uuid::Uuid;
-
 use crate::server::MetricsEndpointConfig;
-use crate::stats::virtual_machine::VirtualMachine;
-
-// Propolis is built and some tests are run on non-illumos systems. The real
-// `kstat` infrastructure cannot be built there, so some conditional compilation
-// tricks are needed
-#[cfg(all(not(test), target_os = "illumos"))]
-use oximeter_instruments::kstat::KstatSampler;
+use crate::vm::InterfaceIdentifiers;
 
+mod network_interface;
 mod pvpanic;
-pub(crate) mod virtual_machine;
-pub use self::pvpanic::PvpanicProducer;
+mod virtual_machine;
 
-// Interval on which we ask `oximeter` to poll us for metric data.
+#[cfg(all(not(test), target_os = "illumos"))]
+use self::network_interface::InstanceNetworkInterfaces;
+pub(crate) use self::pvpanic::PvpanicProducer;
```

Comment:
This structure didn't need to be just `pub`. I updated this to be consistent with the updates here in updating modules and making it cleaner than exposing it through the crate (as we already use `mod` here).